provider/cluster/types/types.go

```diff
@@ -71,3 +75,70 @@ type ServiceLog struct {
 	Stream  io.ReadCloser
 	Scanner *bufio.Scanner
 }
+
+type LeaseEventObject struct {
+	Kind      string `json:"kind" yaml:"kind"`
+	Namespace string `json:"namespace" yaml:"namespace"`
+	Name      string `json:"name" yaml:"name"`
+}
+
+type LeaseEvent struct {
+	Type                string           `json:"type" yaml:"type"`
+	ReportingController string           `json:"reportingController,omitempty" yaml:"reportingController"`
+	ReportingInstance   string           `json:"reportingInstance,omitempty" yaml:"reportingInstance"`
+	Time                time.Time        `json:"time" yaml:"time"`
+	Reason              string           `json:"reason" yaml:"reason"`
+	Object              LeaseEventObject `json:"object" yaml:"object"`
+}
+
+// EventsWatcher
+type EventsWatcher interface {
+	Stop()
+	ResultChan() <-chan *eventsv1.Event
+}
+
+// EventsFeed
+type EventsFeed struct {
```

Comment:
I'd prefer to only expose an interface and implement this with a private struct.

Also, instead of writing to `NomNom()` with the fallback every time, how about a method that does that, like `Write()` or `Send()`?